compiler/qsc_qasm/src/tests/declaration/array.rs

```diff
@@ -4,31 +4,425 @@
 mod bit;
 mod qubit;
 
-use crate::tests::{compile_fragments, fail_on_compilation_errors};
+use crate::tests::{compile_fragments, compile_qasm_to_qsharp, fail_on_compilation_errors};
+use expect_test::expect;
 use miette::Report;
```

Comment:
We should have test cases that 0 length arrays and 0 length sub arrays are supported.